git_machete/cmd.py

```diff
@@ -803,6 +806,74 @@ def slurp_file_or_empty(path: str) -> str:
     save_definition_file()
 
 
+def sync_before_pr(cli_ctxt: CommandLineContext) -> None:
+    global empty_line_status
+
+    expect_at_least_one_managed_branch()
+
+    empty_line_status = True
+
+    def print_new_line(new_status: bool) -> None:
+        global empty_line_status
+        if not empty_line_status:
+            print("")
+        empty_line_status = new_status
+
+    cb = current_branch(cli_ctxt)
+    try:
+        expect_in_managed_branches(cb)
+    except MacheteException:
+        add(cli_ctxt, cb)
+
+    if is_merged_to_upstream(cli_ctxt, cb):
+        return
+    s, remote = get_strict_remote_sync_status(cli_ctxt, cb)
+    statuses_to_sync = (UNTRACKED, AHEAD_OF_REMOTE)
+    needs_remote_sync = s in statuses_to_sync
+
+    if needs_remote_sync:
+        print_new_line(False)
+        sys.stdout.write(f"Checking out {bold(cb)}\n")
```

Comment:
Hmmm is checkout needed? There's an assumption that `cb` is already the current branch 🤔